lib/dialects/mysql/index.js

```diff
@@ -162,16 +162,16 @@ class Client_MySQL extends Client {
   }
 
   async cancelQuery(connectionToKill) {
-    const conn = await this.acquireConnection();
+    const conn = await this.acquireRawConnection();
     try {
-      return await this.query(conn, {
+      return await this._query(conn, {
         method: 'raw',
         sql: 'KILL QUERY ?',
```

Comment:
There is a bit of semantic difference between MySQL and Postgress:

- `KILL QUERY ?` in MySQL stops just the current query, but keeps the transaction active. So the user can send further queries to the same transaction and commit it eventually.
- `SELECT pg_cancel_backend($1)` in Postress stops the current query, and also automaticaly rejects the transaction. So any further queries sent to the same transaction will fail and the transaction will be eventually rolled back.

Should we keep this semantinc difference? Or should we rather manually roll back the transaction in MySQL as well. If we decide either way, it won't be a breaking change, as the query cancellation didn't work in transactions so far anyway.